### ISSUE TITLE
plot: implement more default plots

### ISSUE THREAD

**Issue by pared** [iterative/dvc#3801]
There are some plots that need to be implemented to be default templates:
- [x] default
- [x] scatter
- [x] confusion matrix
- [ ] histogram
- [ ] feature importance
- [ ] bar chart

**Comment by steffansluis** [comment]
Would love some more options here, specifically I was thinking of using a bar chart, histogram or pie chart to visualize my class counts for training samples :bar_chart:. What would be even better would be a stacked bar chart of where the diff between something like class counts is visualized as (red/green/different colored) stacked bars!

**Comment by pared** [comment]
@steffansluis 
So, in case of `plots show` it would be bar chart where x axis are classes, and y is class count, and in case of diff - x axis is revisions, we group bars for particular revisions and differentiate them by color?

**Comment by steffansluis** [comment]
@pared The `plots show` case is correct, classes on the x axis and class count on y. For the diff, I'd say classes on the x axis and stacked bars in the order of revisions, with the colors corresponding to revisions indeed, class counts on the y-axis? I think that is what you meant also?

It might be a bit difficult with regards to when class counts actually go down between revisions, I was only considering the case of comparing two revisions, in that case you could simply stack a red bar if the delta is negative and a green bar if the delta is positive.

If the point is to compare the counts between two revisions, I think a general bar chart with colors corresponding to the revisions makes more sense. I'd say both are valid usecases, depending on whether you are interested in the absolute difference between revisions (e.g. when doing experiments) or the relative differenve (e.g. a PR that adds more training data).

For the diff case, if there is more than 1 commit between the points of comparison, it might even make sense to offers the option of a stacked bar chart with a delta for each of the commits (if there is a difference). The problem there would be that you would need to color to represent the revisions, so you would need some kind of filling pattern (or something else) for the bars to differentiate between positive and negative deltas. (not sure if this is really feasible).

This is mostly my brain taking your comment and running with the options, so please let me know what you think!

TLDR;
-  `plots show`: Regular bar chart, 1 color, classes on the x axis, class counts on the y axis.
-  Diff: Stacked bar chart, classes on the x axis, class counts on the y axis, 3 colors, 1 for the bars, presumably red for negative delta, presumably green for positive delta.
- Compare mutliple revisions: Stacked bar chart, classes on the x axis, class counts on the y axis, 1 color per revision, some kind of way to distinguish between positive and negative delta?

**Comment by pared** [comment]
@steffansluis I see we probably have a different vision of that. What I had in mind was something like that:
https://vega.github.io/vega-lite/examples/bar_grouped.html

Where the `age` is our `rev`.

Though, I have to admit that idea of of `delta` bar chart sounds interesting. What if we diff 3 `revisions`? Plot original bar chart for most recent and deltas for other revisions?

**Comment by steffansluis** [comment]
@pared Wouldn't grouping by revision make it harder to see the difference between revisions? I would imagine you would want to group them by the class and put the revisions where the `Male/Female` is, so that you can easily see how a single count changed between revisions.

For the delta bar chart indeed I'd go with the most recent revision as the 'base' and deltas for the other revisions.
